### PR TITLE
📖 book: Add note about Kubernetes v1.23.{0-2} CRD issue and ClusterClass

### LINKED ISSUE #5990: Add a note to versions.md about ClusterClass issues on Kubernetes v1.23.{0,1,2}
There was an announcement that there is an issue with Kubernetes v1.23.{0,1,2} and CRDs using `x-kubernetes-preserve-unknown-fields: true` (https://groups.google.com/a/kubernetes.io/g/dev/c/Xl1sm-CItaY https://github.com/kubernetes/kubernetes/issues/107690)

ClusterAPI uses preserve unknown fields only for ClusterClass patches and variables:
* ClusterClass: jsonPatches[x].value, openAPIV3Schema.{default,enum,example}
* Cluster: variables[x].value, variables.overrides[x].value

A concrete example. When the following YAML is deployed:
```yaml
  - name: nestedArrayVar
    required: true
    schema:
      openAPIV3Schema:
        type: array
        items:
          type: object
          properties:
            nestedField:
              type: string
        default:
        - nestedField: "abc"
        - nestedField: "def"
```
the apiserver will prune nested fields in the default value:
```yaml
  - name: nestedArrayVar
    required: true
    schema:
      openAPIV3Schema:
        default:
        - {}
        - {}
        items:
          properties:
            nestedField:
              type: string
          type: object
        type: array
```
This can happen with all other fields mentioned above. 

Because of this issue it's not recommended to use ClusterClass patches and variables on a management cluster with Kubernetes v1.23.{0,1,2}.

Let's add a **short** note to versions.md (https://cluster-api.sigs.k8s.io/reference/versions.html) that it's not recommened to use ClusterClass with patches on those Kubernetes versions and link back to the current issue.

/kind feature

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5990

/cherry-pick release-1.1
/assign @fabriziopandini

### CODE REVIEW

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -52,7 +52,10 @@ These diagrams show the relationships between components in a Cluster API releas
...
 | Kubernetes v1.21 | ✓                                | ✓                             | ✓                                | ✓                             | ✓                                | ✓                             |
 | Kubernetes v1.22 |                                  | ✓                             | ✓                                | ✓                             | ✓                                | ✓                             |
-| Kubernetes v1.23 |                                  |                               | ✓                                | ✓                             | ✓                                | ✓                             |
+| Kubernetes v1.23* |                                  |                               | ✓                                | ✓                             | ✓                                | ✓                             |
+
+\* There is an issue with CRDs in Kubernetes v1.23.{0-2}. ClusterClass with patches is affected by that (for more details please see [this issue](https://github.com/kubernetes-sigs/cluster-api/issues/5990)). Therefore we recommend to use Kubernetes v1.23.3+ with ClusterClass.
```
Comment (fabriziopandini) [inline docs/book/src/reference/versions.md:57]: Q: should the the note go at the end of the paragraph

Comment (fabriziopandini) [inline docs/book/src/reference/versions.md:57]: Also, should the header of the table have CAPI v1beta1 (v1.0)  --> CAPI v1beta1 (v1.{0,1}) (or v1.x)
Not strictly related to this PR, but useful for the release...

**Review by fabriziopandini — COMMENTED** [review]
Two small nits, also it seems that some links are broken (also this probably unrelated to the PR)

File: docs/book/src/reference/versions.md (hunk 1 above)
Comment (fabriziopandini) [reply docs/book/src/reference/versions.md:57]: Also, should the header of the table have CAPI v1beta1 (v1.0) --> CAPI v1beta1 (v1.{0,1}) (or v1.x)
Not strictly related to this PR, but useful for the release...

Comment (sbueringer) [reply docs/book/src/reference/versions.md:57]: > Q: should the the note go at the end of the paragraph

I assume you mean below l.60? I though it's good to have it directly under the table as the table is referring to it.
But I'm fine to change it if you prefer having it below.

You're right about the table header. I'll probably change it as suggested if I can't come up with something better :)

While it's great to have this overview, I wonder if it's confusing to users if they see a different version of this table in every book (and I assume some have conflicting information as we don't really maintain old versions).
So we  maybe should consider some changes (like e.g. the book of each version only contains the information which Kubernetes versions are supported in that release). 
But as that's probably not a quick discussion, let's ignore that for now.

Comment (sbueringer) [reply docs/book/src/reference/versions.md:57]: I went with 1.x and also adjusted some occurrences in the text above.

Let me know if you would prefer having the note further down

**Pull request merged**